snuba/datasets/errors_replacer.py

```diff
@@ -145,25 +150,34 @@ def get_count_query(self, table_name: str) -> Optional[str]:
         return self.count_query_template % args
 
 
-def get_project_exclude_groups_key(
+def get_project_exclude_groups_key_and_type_key(
     project_id: int, state_name: Optional[ReplacerState]
-) -> str:
-    return f"project_exclude_groups:{f'{state_name.value}:' if state_name else ''}{project_id}"
+) -> Tuple[str, str]:
+    key = f"project_exclude_groups:{f'{state_name.value}:' if state_name else ''}{project_id}"
+    return key, f"{key}-type"
 
 
 def set_project_exclude_groups(
-    project_id: int, group_ids: Sequence[int], state_name: Optional[ReplacerState]
+    project_id: int,
+    group_ids: Sequence[int],
+    state_name: Optional[ReplacerState],
+    replacement_type: str,
 ) -> None:
-    """Add {group_id: now, ...} to the ZSET for each `group_id` to exclude,
+    """
+    Add {group_id: now, ...} to the ZSET for each `group_id` to exclude,
     remove outdated entries based on `settings.REPLACER_KEY_TTL`, and expire
-    the entire ZSET incase it's rarely touched."""
+    the entire ZSET incase it's rarely touched.
+
+    Add replacement type for this replacement.
+    """
 
     now = time.time()
-    key = get_project_exclude_groups_key(project_id, state_name)
+    key, type_key = get_project_exclude_groups_key_and_type_key(project_id, state_name)
     p = redis_client.pipeline()
 
     group_id_data: Mapping[str, float] = {str(group_id): now for group_id in group_ids}
     p.zadd(key, **group_id_data)
```

Comment:
What happens if you execute two replacements for the same project of different type?
This line adds the groups to exclude to a set. The set may already exist if we already executed a replacement on the same project. 
Though the tag key you are setting below would replace the existing one. This means you would be loosing track of the previous replacement type.
Instead of `p.set(type_key, replacement_type)` you may want to keep track of multiple types just like we do for keeping track of multiple project with a sorted set.